### ISSUE TITLE
Change the launcher's (too) bright green colors 

### ISSUE THREAD

**Issue by ltGuillaume** [TBog/TBLauncher#72]
The main app's toxic green gives it a bit of an old terminal/XT computer vibe. I'm all for retro, but it does mean it doesn't blend with the rest of the OS. Therefore, it would be fantastic if the app would either 1) follow the accent color set in Android's `Settings` -> `Display` -> `Styles & Wallpapers`, or 2) allow to set the accent color and the background color for TBLauncher's menus and settings manually.

I guess the next step would be to support dark mode when set in Android.

As I understand, the light green background is partly there because there's no border around the set color's preview (so they could disappear when e.g. white is selected). Perhaps the following can be used to easily set a border around the color preview: https://stackoverflow.com/a/3264140

**Comment by TBog** [comment]
* The green you speak of blends with my phone perfectly (Samsung has OS themes), but I understand what you want.
* I don't know how to find the color you set in `Settings` -> `Display` -> `Styles & Wallpapers`. Also I'm not sure what you're talking about, I don't have any menu like that on my Samsung phone and the Android 10 emulator doesn't have a color option, only Dark mode

From what I understand you want an option to change the green color of the popups and dialogs
popup: 
![image](https://user-images.githubusercontent.com/6465804/103245037-8c587a80-4967-11eb-8570-724ddd306e7c.png)

dialog: 
![image](https://user-images.githubusercontent.com/6465804/103245005-6af78e80-4967-11eb-8e56-7d2020075423.png)

Is that all you need when you say `(too) bright green colors`?

**Comment by ltGuillaume** [comment]
> I don't know how to find the color you set in `Settings` -> `Display` -> `Styles & Wallpapers`.

Librechair does it like this: https://github.com/linsui/librechair/blob/c999d70cf2a57fb7243159748f23b49482a6d8d7/lawnchair/src/ch/deletescape/lawnchair/colors/colorResolvers.kt#L54

> Also I'm not sure what you're talking about, I don't have any menu like that on my Samsung phone and the Android 10 emulator doesn't have a color option, only Dark mode

You're right, sorry, it was put behind the Developer options later in development, apparently it was LineageOS that put it where I said it was. But it's still available on any Android 10 device, see https://www.androidauthority.com/change-system-accent-color-android-10-1025163/

> The green you speak of blends with my phone perfectly (Samsung has OS themes), but I think I understand what you want.
> From what I understand you want an option to change the green color of the popups and dialogs
> 
>     1. popup text color:
>        ![image](https://user-images.githubusercontent.com/6465804/103245037-8c587a80-4967-11eb-8570-724ddd306e7c.png)
> 
>     2. dialog border:
>        ![image](https://user-images.githubusercontent.com/6465804/103245005-6af78e80-4967-11eb-8e56-7d2020075423.png)
> 
> 
> Is that all you need when you say `(too) bright green colors`?

Those colors 1 and 2 indeed, but also 3 the Settings background/accent colors and 4 the highlighted (ripple) color of #20.

Simply put, I'd like to get something like a neutral grey color (such as #eeeeee) to be the overall accent color where the green is now.

**Comment by TBog** [comment]
The ripple color has its own issue for now.

I don't think a color chooser for the color chooser (Settings activity) is needed. 
1. I'll change the default Settings background to black/dark and the text to green (now that I figured out how to overwrite the generated preference menu).
2. I'll think of a way to use the colors from the result list preferences to also change the settings. For example use the "Result list - background color" as the Settings background and the "Result list - text color" for the text.

What do you think?

**Comment by ltGuillaume** [comment]
>2. I'll think of a way to use the colors from the result list preferences to also change the settings. For example use the "Result list - background color" as the Settings background and the "Result list - text color" for the text.

That would make more sense. The green color isn't neutral/doesn't fit the OS, and a color like it by default may deter people from starting to use TBLauncher in the first place, which would be a pity. If it's customizable, that's a great step. And I suggested using the system's accent color for if you want to make a better first impression towards new users.

**Comment by TBog** [comment]
I'll add the system's accent color as a default color.
But I'm not sure the amount of work needed to change the colors of the Settings activity is warranted.
What if the user chooses a combination of colors that is too hard to read? This would block the user from changing them back.

**Comment by ltGuillaume** [comment]
> The Settings activity generates the layout from an XML file and I can only set the theme of the activity. It's not possible AFAIK to make/change a theme programmatically and I don't want to change the way the preferences are currently handled.
Ah that could be the case, I don't know about the details of customizing the theme programmatically. But if it's just a listView, I guess you could simply set the text color in (and setTint() for drawables like a color preview border) inside getView().

**Comment by TBog** [comment]
> Ah that could be the case, I don't know about the details of customizing the theme programmatically. But if it's just a listView, I guess you could simply set the text color in (and setTint() for drawables like a color preview border) inside getView().

All I do is `setContentView(R.layout.activity_settings);` in `onCreate` I have limited control of the generated views

**Comment by TBog** [comment]
Like I said, I'll research some more when I have time and other issues are less important.
Right now the only fix I can think of for the Settings activity is to change the background to something dark or black.

**Comment by ltGuillaume** [comment]
I've been searching for a solution, but the only thing I could find was to set up different themes for the different accent colors, so setting something like `textColor` and `colorAccent` programmatically seems impossible for PreferenceFragmentCompat. So that's definitely not worth it.

**Comment by TBog** [comment]
I'm starting to heavily dislike the theme api that is used by the Preferences.
I'm still working on this. Would you be willing to help test a preview build? I'm not the best judge for chromatic changes.

**Comment by TBog** [comment]
There is a new release [v5.4-colors.2](https://github.com/TBog/TBLauncher/releases/tag/v5.4-colors.2) with
1. settings activity themes 
2. customizable popup colors
3. ripple color

Need some feedback and help choosing colors for the settings activity themes.
![Screenshot_1610649464](https://user-images.githubusercontent.com/6465804/104633858-75749080-56a8-11eb-8893-d34c316e3165.png)

Here is a preview of the popup with yellow and red
![Screenshot_1610649390](https://user-images.githubusercontent.com/6465804/104633715-3cd4b700-56a8-11eb-8597-ccff01396cc6.png)

**Comment by ltGuillaume** [comment]
Great stuff! I'll check it out very soon!

**Comment by ltGuillaume** [comment]
What I've seen so far:
1. The changed highlight color is not applied to "..." (for the settings popup) next to the search input
2. Would be nice to have a color setting for the highlight color, bright green titles, the secondary text and the scroll bar in the settings, but that's obviously very low priority.

**Comment by TBog** [comment]
> 1. The changed highlight color is not applied to the button to the left and the one to the right "⋮" (settings popup) of the search input.

Will fix. 

> 2. Would be nice to have a color setting for the highlight color, bright green titles, the secondary text and the scroll bar in the settings, but that's obviously very low priority.

Can't have customized colors for the Settings activity.
What I can do is add more themes besides the 3 (Black, White, Dark) that are now.
Please help with colors. I don't know what combination to use.

**Comment by ltGuillaume** [comment]
Compared with the settings I have elsewhere, I'd have these colors for Settings:
- Primary text/titles: #e0e0e0
- Secondary text/scroll bar: #9e9e9e

Not sure if it looks great right away.

**Comment by TBog** [comment]
@ltGuillaume can you please look at the [latest pre-release](https://github.com/TBog/TBLauncher/releases/tag/v5.4-colors.4) and give some feedback?

**Comment by ltGuillaume** [comment]
(1) seems to be fixed! Nice!

I see no differences regarding (2), but I think that's 

3. Also, the lines underneath headers in popups cut off the lower part of the title's characters (with the `g` in Settings and Widget).

**Comment by TBog** [comment]
> I see no differences regarding (2), but I think that's OK.

The scrollbar? Which one? 

> 3. Also, the lines underneath headers in popups cut off the lower part of the title's characters (with the `g` in Settings and Widget).

if I change the draw order and have the text over the bar would the overlap be ok?

**Comment by ltGuillaume** [comment]
> > I see no differences regarding (2), but I think that's OK.
> 
> You're talking about the scrollbar? Which one?
No I meant that I don't know if you've changed anything to (the colors in) the Settings. 

> > 3. Also, the lines underneath headers in popups cut off the lower part of the title's characters (with the `g` in Settings and Widget).
> 
> If I change the draw order and have the text over the bar would the overlap be ok?

I think it would still look very messy tbh.

**Comment by TBog** [comment]
The `White` theme in the setting is monochrome now, as in only black, white and shades of gray.
It took some time to figure out how to change every color and remove the green.

Next I'll make the black theme monochrome and maybe add others.

Could you also check the other dialog and popup windows?

**Comment by ltGuillaume** [comment]
Aah yes, I tried all three for the first time with this version, whereas I only saw the dark and black ones with the previous. White looks nice though!

What I've seen so far:

1. The popup when long-pressing an app has a scroll bar, which seem unnecessary. Moreover, this scroll bar goes "outside" of the popup due to the rounded corners. The colors are correct.
2. The Rename App and Add Tag dialogs are still green (border, title and the buttons "cancel" and "rename").

**Comment by TBog** [comment]
With the white theme the rename dialog should look like this ![SmartSelect_20210118-214549_TinyBit launcher.jpg](https://user-images.githubusercontent.com/6465804/104957346-a7526380-59d6-11eb-8bf9-8873c89f2a9f.jpg)

**Comment by ltGuillaume** [comment]
Aah that makes sense. I set the theme to white and I'm seeing this:
![image](https://user-images.githubusercontent.com/4777345/104957654-d9f85e00-59ce-11eb-92c6-b944b430df68.png)
I'd say the gradient doesn't do it much good, gives an early 2000s feel.

**Comment by TBog** [comment]
@ltGuillaume can we close this issue with the current state of [Feature - menu color preferences (6)](https://github.com/TBog/TBLauncher/releases/tag/v5.4-colors.6)

**Comment by ltGuillaume** [comment]
Fantastic! Thank you, great work!

**Comment by TBog** [comment]
Cursor color not changed.
Possible fix: https://www.tfzx.net/article/2715806.html
or: https://stackoverflow.com/questions/35583320/changing-androids-edittext-cursor-pointer-color-programmatically?noredirect=1&lq=1

**Comment by ltGuillaume** [comment]
> 
> 
> Cursor color not changed.
> Possible fix: https://www.tfzx.net/article/2715806.html
> or: https://stackoverflow.com/questions/35583320/changing-androids-edittext-cursor-pointer-color-programmatically?noredirect=1&lq=1

Apart from this, I just found out that the "Do you want to create a shortcut" dialog is still green.

**Comment by TBog** [comment]
> I just found out that the "Do you want to create a shortcut" dialog is still green.

Yes, forgot about it, I'm mostly testing on API 24 and that dialog is not used.
Thanks for the reminder.

**Comment by ltGuillaume** [comment]
Aaah that makes sense, yes. Android 7.0 is still my testing ground as well, especially since the AVD is much cleaner and faster.

**Comment by ltGuillaume** [comment]
I just realized that the cursor "teardrop" in the Rename App dialog isn't green, unlike in the search bar, although its transparency has been corrupted (there's a black square around it).

**Comment by ltGuillaume** [comment]
> 
> 
> I just realized that the cursor "teardrop" in the Rename App dialog isn't green, unlike in the search bar, although its transparency has been corrupted (there's a black square around it).

For https://github.com/TBog/TBLauncher/releases/tag/v5.4-modes.1

- Cursor "teardrop" color in search isn't green anymore, don't know which color it now uses (secondary text color perhaps?). However, the **cursor seems to have changed to yellow** since this build
- I think both the rename dialog and the change icon dialog now adhere to the Preferences theme, instead of the colors & corner radius setting you can set for popups, right? 

So, what remains to my knowledge is:
1. "Do you want to create a shortcut" dialog is still green
2. Cursor in search now is yellow (could be some standard Android color, as the default Android color to highlight a row in my DroidShows app is also this yellow).

**Comment by TBog** [comment]
> For https://github.com/TBog/TBLauncher/releases/tag/v5.4-modes.1
> 
> - Cursor "teardrop" color in search isn't green anymore, don't know which color it now uses (secondary text color perhaps?). However, the **cursor seems to have changed to yellow** since this build.

It's the color set for ripple in the results list. Should this be customized separately? 

> - I think both the rename dialog and the change icon dialog now adhere to the Preferences theme, instead of the colors & corner radius setting you can set for popups, right? But maybe it still uses some of the (text) colors set by the user? Judging from the following screenshot, changing the settings theme for a moment to white has strange effects:
> ![image](https://user-images.githubusercontent.com/4777345/106676294-dcee7380-65b6-11eb-96f6-c3957c9e5e1f.png)

I don't understand what is wrong.

> So, what remains to my knowledge is:
> 1. "Do you want to create a shortcut" dialog is still green

Yes

> 2. Cursor in search now is yellow (could be some standard Android color, as the default Android color to highlight a row in my DroidShows app is also this yellow).

Should be the result list ripple color, same color is used for the Search icons ripple and the text highlight. We could make them separate.

**Comment by ltGuillaume** [comment]
> I don't understand what is wrong.

Since it's a popup, I would assume for the Rename and Change Icon dialogs that the popup colors and corner radius setting would be used. And the text is difficult to read on this light settings theme + shadow makes it looks like I'm drunk. 

It was like a popup before, and I'd say the current implementation is quite a step backwards in terms of consistency/unity within the app.

> Should be the result list ripple color, same color is used for the Search icons ripple and the text highlight. We could make them separate.

I guess separating them would make sens, since they're "borrowed" from another element then.

But here's a screenshot while the results list touch color is set to #e0e0e0: 
![image](https://user-images.githubusercontent.com/4777345/106677344-e678db00-65b8-11eb-861d-237665ca309b.png)

I can't seem to find anything that uses the setting "Action icon color" by the way, what is it used for?

> It's the color set for ripple in the results list. Should this be customized separately?

Not necessarily, it's just a bit inconsistent, that's all:
![image](https://user-images.githubusercontent.com/4777345/106677183-9ac63180-65b8-11eb-9fd0-4f264a82c97e.png)
![image](https://user-images.githubusercontent.com/4777345/106677215-a87bb700-65b8-11eb-9eaa-6ed34e810fef.png)

**Comment by TBog** [comment]
> I can't seem to find anything that uses the setting "Action icon color" by the way, what is it used for?

It's used for the contact call and message icon

**Comment by TBog** [comment]
> It was (more) like the popup before, and I'd say the current implementation is quite a step backwards in terms of consistency/unity within the app.

The Settings Activity can't be customized, it just has a couple of themes.

The QuickList dialog can be opened from the Settings and from the long-press pop-up in the QuickList. Initially it was only accessible from the settings and inherit the theme.

Help me create a Settings theme that you'd want to see for the custom icon dialog and QuickList editor.

**Comment by ltGuillaume** [comment]
Simply because it's presented as a popup, not as a Settings screen: they are smaller dialogs on top of the activity you're in. This in contract to the Settings, which is a "full-screen" different Activity. That's why it's confusing that the (let's say "context menu") popups and these popups are so different.

**Comment by ltGuillaume** [comment]
As for a Settings theme: these are the values I have for my popups, not sure if they translate nicely to a full-screen settings activity.

![image](https://user-images.githubusercontent.com/4777345/106680434-eaa7f700-65be-11eb-8cf3-3667f0705342.png)

Corner radius: 0
Border: FF9E9E9E
Background: CC212121
Touch: 424242
Text: E0E0E0
Title: EEEEEE

**Comment by TBog** [comment]
Now that I added the default theme, can we close this?

**Comment by ltGuillaume** [comment]
No! 😛

1. Edit tags still shows a green line around existing tags
1. Edit tags still has an old [+] button
